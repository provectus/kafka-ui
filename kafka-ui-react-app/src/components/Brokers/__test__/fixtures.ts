import { BrokersLogdirs } from 'generated-sources';

export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com', port: 9092 },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com', port: 9092 },
];

export const clusterStatsPayload = {
  brokerCount: 2,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

const partition = {
  broker: 2,
  offsetLag: 0,
  partition: 2,
  size: 0,
};
const topics = {
  name: '_confluent-ksql-devquery_CTAS_NUMBER_OF_TESTS_59-Aggregate-Aggregate-Materialize-changelog',
  partitions: [partition],
};

export const brokerLogDirsPayload: BrokersLogdirs[] = [
  {
    error: 'NONE',
    name: '/opt/kafka/data-0/logs',
    topics: [
      {
        ...topics,
        partitions: [partition, partition, partition],
      },
      topics,
      {
        ...topics,
        partitions: [],
      },
    ],
  },
];
