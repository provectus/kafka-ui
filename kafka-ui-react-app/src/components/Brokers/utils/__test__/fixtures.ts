import { BrokerMetrics } from 'generated-sources';

export const brokerMetricsPayload: BrokerMetrics = {
  segmentSize: 23,
  segmentCount: 23,
  metrics: [
    {
      name: 'TotalFetchRequestsPerSec',
      labels: {
        canonicalName:
          'kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics',
      },
      value: 10,
    },
    {
      name: 'ZooKeeperRequestLatencyMs',
      value: 11,
    },
    {
      name: 'RequestHandlerAvgIdlePercent',
    },
  ],
};
export const transformedBrokerMetricsPayload =
  '{"segmentSize":23,"segmentCount":23,"metrics":[{"name":"TotalFetchRequestsPerSec","labels":{"canonicalName":"kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics"},"value":10},{"name":"ZooKeeperRequestLatencyMs","value":11},{"name":"RequestHandlerAvgIdlePercent"}]}';
