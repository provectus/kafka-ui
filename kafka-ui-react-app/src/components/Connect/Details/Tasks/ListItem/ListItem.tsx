import React from 'react';
import { useParams } from 'react-router-dom';
import { Task, TaskId } from 'generated-sources';
import { ClusterName, ConnectName, ConnectorName } from 'redux/interfaces';
import Dropdown from 'components/common/Dropdown/Dropdown';
import DropdownItem from 'components/common/Dropdown/DropdownItem';
import VerticalElipsisIcon from 'components/common/Icons/VerticalElipsisIcon';
import * as C from 'components/common/Tag/Tag.styled';
import getTagColor from 'components/common/Tag/getTagColor';

interface RouterParams {
  clusterName: ClusterName;
  connectName: ConnectName;
  connectorName: ConnectorName;
}

export interface ListItemProps {
  task: Task;
  restartTask(
    clusterName: ClusterName,
    connectName: ConnectName,
    connectorName: ConnectorName,
    taskId: TaskId['task']
  ): Promise<void>;
}

const ListItem: React.FC<ListItemProps> = ({ task, restartTask }) => {
  const { clusterName, connectName, connectorName } = useParams<RouterParams>();

  const restartTaskHandler = React.useCallback(async () => {
    await restartTask(clusterName, connectName, connectorName, task.id?.task);
  }, [restartTask, clusterName, connectName, connectorName, task.id?.task]);

  return (
    <tr>
      <td>{task.status?.id}</td>
      <td>{task.status?.workerId}</td>
      <td>
        <C.Tag color={getTagColor(task.status)}>{task.status.state}</C.Tag>
      </td>
      <td>{task.status.trace || 'null'}</td>
      <td style={{ width: '5%' }}>
        <div>
          <Dropdown label={<VerticalElipsisIcon />} right>
            <DropdownItem onClick={restartTaskHandler} danger>
              <span>Restart task</span>
            </DropdownItem>
          </Dropdown>
        </div>
      </td>
    </tr>
  );
};

export default ListItem;
