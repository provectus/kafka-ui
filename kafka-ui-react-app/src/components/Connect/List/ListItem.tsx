import React from 'react';
import { FullConnectorInfo } from 'generated-sources';
import { clusterConnectConnectorPath, clusterTopicPath } from 'lib/paths';
import { ClusterName } from 'redux/interfaces';
import { Link, NavLink } from 'react-router-dom';
import { Tag } from 'components/common/Tag/Tag.styled';
import { TableKeyLink } from 'components/common/table/Table/TableKeyLink.styled';
import getTagColor from 'components/common/Tag/getTagColor';
import { useDeleteConnector } from 'lib/hooks/api/kafkaConnect';
import { Dropdown, DropdownItem } from 'components/common/Dropdown';
import { useConfirm } from 'lib/hooks/useConfirm';

import * as S from './List.styled';

export interface ListItemProps {
  clusterName: ClusterName;
  connector: FullConnectorInfo;
}

const ListItem: React.FC<ListItemProps> = ({
  clusterName,
  connector: {
    name,
    connect,
    type,
    connectorClass,
    topics,
    status,
    tasksCount,
    failedTasksCount,
  },
}) => {
  const confirm = useConfirm();
  const deleteMutation = useDeleteConnector({
    clusterName,
    connectName: connect,
    connectorName: name,
  });

  const handleDelete = () => {
    confirm(
      <>
        Are you sure want to remove <b>{name}</b> connector?
      </>,
      async () => {
        await deleteMutation.mutateAsync();
      }
    );
  };

  const runningTasks = React.useMemo(() => {
    if (!tasksCount) return null;
    return tasksCount - (failedTasksCount || 0);
  }, [tasksCount, failedTasksCount]);

  return (
    <tr>
      <TableKeyLink>
        <NavLink to={clusterConnectConnectorPath(clusterName, connect, name)}>
          {name}
        </NavLink>
      </TableKeyLink>
      <td>{connect}</td>
      <td>{type}</td>
      <td>{connectorClass}</td>
      <td>
        <S.TagsWrapper>
          {topics?.map((t) => (
            <Tag key={t} color="gray">
              <Link to={clusterTopicPath(clusterName, t)}>{t}</Link>
            </Tag>
          ))}
        </S.TagsWrapper>
      </td>
      <td>
        {status && <Tag color={getTagColor(status.state)}>{status.state}</Tag>}
      </td>
      <td>
        {runningTasks && (
          <span>
            {runningTasks} of {tasksCount}
          </span>
        )}
      </td>
      <td>
        <div>
          <Dropdown>
            <DropdownItem onClick={handleDelete} danger>
              Remove Connector
            </DropdownItem>
          </Dropdown>
        </div>
      </td>
    </tr>
  );
};

export default ListItem;
