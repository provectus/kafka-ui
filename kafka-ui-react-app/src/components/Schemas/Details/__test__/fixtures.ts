import { SchemaSubject } from 'generated-sources';

export const schema: SchemaSubject = {
  subject: 'test',
  version: '1',
  id: 1,
  schema:
    '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
  compatibilityLevel: 'BACKWARD',
};

export const versions: SchemaSubject[] = [
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
  },
  {
    subject: 'test',
    version: '2',
    id: 2,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
  },
];
