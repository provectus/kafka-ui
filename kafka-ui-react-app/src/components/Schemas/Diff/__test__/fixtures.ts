import { SchemaSubject, SchemaType } from 'generated-sources';

export const versions: SchemaSubject[] = [
  {
    subject: 'test',
    version: '3',
    id: 3,
    schema:
      'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.PROTOBUF,
  },
  {
    subject: 'test',
    version: '2',
    id: 2,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
];
