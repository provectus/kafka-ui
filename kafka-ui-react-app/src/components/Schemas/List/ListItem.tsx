import React from 'react';
import { SchemaSubject } from 'generated-sources';
import { NavLink } from 'react-router-dom';

export interface ListItemProps {
  subject: SchemaSubject;
}

const ListItem: React.FC<ListItemProps> = ({
  subject: { subject, version, compatibilityLevel },
}) => {
  return (
    <tr>
      <td>
        <NavLink
          exact
          to={`schemas/${subject}/latest`}
          activeClassName="is-active"
          className="title is-6"
        >
          {subject}
        </NavLink>
      </td>
      <td>{version}</td>
      <td>
        <span className="tag is-link">{compatibilityLevel}</span>
      </td>
    </tr>
  );
};

export default ListItem;
