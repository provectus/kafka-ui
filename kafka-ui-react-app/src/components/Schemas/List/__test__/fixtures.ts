import {
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';

const schemas = [
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
];

export const schemasPayload = {
  pageCount: 1,
  schemas,
};

export const schemasEmptyPayload = {
  pageCount: 1,
  schemas: [],
};
