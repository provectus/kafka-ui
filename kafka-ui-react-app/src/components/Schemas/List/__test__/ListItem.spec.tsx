import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { schemas } from './fixtures';
import ListItem from '../ListItem';

describe('ListItem', () => {
  it('renders schemas', () => {
    const wrapper = mount(
      <Router>
        <ListItem subject={schemas[0]} />
      </Router>
    );

    expect(wrapper.find('NavLink').length).toEqual(1);
    expect(wrapper.find('td').length).toEqual(3);
  });

  it('matches snapshot', () => {
    expect(
      mount(
        <Router>
          <ListItem subject={schemas[0]} />
        </Router>
      )
    ).toMatchSnapshot();
  });
});
