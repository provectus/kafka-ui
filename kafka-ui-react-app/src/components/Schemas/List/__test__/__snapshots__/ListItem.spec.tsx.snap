// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ListItem
      subject={
        Object {
          "compatibilityLevel": "BACKWARD",
          "id": 1,
          "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord1\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
          "subject": "test",
          "version": "1",
        }
      }
    >
      <tr>
        <td>
          <NavLink
            activeClassName="is-active"
            className="title is-6"
            exact={true}
            to="schemas/test/latest"
          >
            <Link
              aria-current={null}
              className="title is-6"
              to={
                Object {
                  "hash": "",
                  "pathname": "/schemas/test/latest",
                  "search": "",
                  "state": null,
                }
              }
            >
              <LinkAnchor
                aria-current={null}
                className="title is-6"
                href="/schemas/test/latest"
                navigate={[Function]}
              >
                <a
                  aria-current={null}
                  className="title is-6"
                  href="/schemas/test/latest"
                  onClick={[Function]}
                >
                  test
                </a>
              </LinkAnchor>
            </Link>
          </NavLink>
        </td>
        <td>
          1
        </td>
        <td>
          <span
            className="tag is-link"
          >
            BACKWARD
          </span>
        </td>
      </tr>
    </ListItem>
  </Router>
</BrowserRouter>
`;
